### PR TITLE
HDS-1889: Get Keycloak api tokens

### PR DESCRIPTION
**Pull request by NikoHelle**
## Description

Keycloak api tokens must be fetched one by one and with extra parameters.

Added new props for those and a way to get tokens one by one. If a fetch fails, other tokens are not fetched and already fetched ones are discarded. There are no API changes, only props. That is why documentation changed only a little.

Had to fix an issue with `httpPoller` and `httpPollerWithPromises`. If a request is aborted, the error was not passed to the caller.

To know what server was used for login, the callback urls must differ. Keycloak's callback is simply `callback_kc.html`.  The server type is also in storybook arguments as "useKeycloak:true/false"

## Related Issue

Closes [HDS-1889](https://helsinkisolutionoffice.atlassian.net/browse/HDS-1889)

## How Has This Been Tested?

Local unit tests.

[Demo](https://city-of-helsinki.github.io/hds-demo/login/?path=/story/components-login--callback)
[Docs demo](https://city-of-helsinki.github.io/hds-demo/login-docs/components/login-docs/)

Note: Missing comma in the demo "...is Tunnistamo This can...." is fixed in code and visible in Loki images. Demo update requires resetting all loginProp urls, so did not re-create demo for that. 😄 

## Screenshots (if appropriate):

[HDS-1889]: https://helsinkisolutionoffice.atlassian.net/browse/HDS-1889?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: site/src/docs/components/login/index.mdx [hunk 1]
```diff
@@ -154,17 +154,43 @@ Detailed information is in the <UsagePageAnchorLink anchor="signals">signals sec
...
+    redirect_uri: `YOUR_REGISTERED_REDIRECT_URL`,
+    silent_redirect_uri: `URL_TO_YOUR_REGISTERED_LOCAL_SILENT_RENEW.HTML`,
+    post_logout_redirect_uri: `YOUR_REGISTERED_LOGOUT_URL`,
   },
   apiTokensClientSettings: { url: 'https://tunnistamo.dev.hel.ninja/api-tokens/' },
   sessionPollerSettings: { pollIntervalInMs: 10000 },
 };
 ```
+
+If you are using Keycloak, then use these settings:
```
Comment (harriplappalainen) [inline site/src/docs/components/login/index.mdx:174]: If Keycloak will be replacing Tunnistamo, should we add a notification about that somewhere in docs? Maybe in Index page, Introduction section?

**Review by harriplappalainen — COMMENTED** [review]
Gooood work! 👍 Hard to say anything about the overall logic in these but the code seems fine and login is working so that's good enough for me 😅

**Review by laurakarhu — APPROVED** [review]
Seems to be working. I agree with Harri, that mentioning Tunnistamo will be replaced later on is a good idea.

**Review by mrTuomoK — APPROVED** [review]
Not an expert on this, but should work. Again, gooooood work 👍

**Pull request merged**